.github/workflows/build-test.yml

```diff
@@ -20,12 +20,13 @@ jobs:
     runs-on:                    ${{ matrix.platform }}
     steps:
       - name:                   Checkout sources
-        uses:                   actions/checkout@v2
+        uses:                   actions/checkout@master
         with:
-          fetch-depth:          50
-      - name:                   Checkout submodules
+          submodules:           true
+      # https://github.com/actions/cache/issues/133 
+      - name:                   Fixup the owner of ~/.cargo/
         shell:                  bash
-        run:                    git submodule update --init --recursive
+        run:                    bash sudo chown -R $(whoami):$(id -ng) ~/.cargo/
```

Comment:
This will try to run `sudo` as shell script I believe. Also windows doesn't have `chown`, `whoami` and `id`. The permission system on Windows works quite different.

Reply:
I expected running `bash` on Windows would use the `Windows Subsystem for Linux 2`, but maybe I'm wrong.

Reply:
`bash` on windows is just a recompilation of `bash` for windows I believe. You have to use `wsl` to run commands on `Windows Subsystem for Linux`. I assume it is not enabled on gh actions though.